### PR TITLE
pkg/scaffold/ansible; Add k8s_status module

### PR DESCRIPTION
**Pull request by dymurray**
**Description of the change:**
Add k8s_status Ansible module into the build for Ansible Operator

**Motivation for the change:**
This Ansible module is needed for the managed status work in Ansible Operator

### CODE REVIEW

File: pkg/scaffold/ansible/k8s_status.go [hunk 1]
```diff
@@ -0,0 +1,416 @@
...
+
+description:
+  - Sets the status field on a Kubernetes API resource. Only should be used if you are using Ansible to
+    implement a controller for the resource being modified.
+
+options:
+  status:
+    type: dict
+    description:
+    - A object containing ` + "`key: value`" + ` pairs that will be set on the status object of the specified resource.
```
Comment (dymurray) [inline pkg/scaffold/ansible/k8s_status.go]: I know how weird this looks.... but the python file has backticks in it. This was the cleanest way I could get the entire file to be a single string.

**Comment by fabianvf** [comment]
+1 to `k8s_status` test, you should be able to add the task to update the status to `test/ansible-memcached/tasks.yml`, and an assertion that the field exists in the status to `test/ansible-memcached/tasks.yml` (just add it anywhere before the CR is deleted). On line 26 of the `asserts.yml` you can see an example of a task that examines the status of an object.

**Comment by dymurray** [comment]
I don't believe Ansible CI will pass until https://github.com/operator-framework/operator-sdk/issues/958 is fixed. The image we are pulling from does not include the k8s status module I'm trying to add in this PR.

File: test/ansible-memcached/asserts.yml [hunk 2]
```diff
@@ -45,6 +45,21 @@
...
 
+    - name: debug CR lookup
+      debug:
+        var: cr
+      vars:
+        deploy: '{{ lookup("k8s",
+          kind=custom_resource.kind,
+          api_version=custom_resource.apiVersion,
+          namespace=namespace,
+          name=custom_resource.metadata.name
```
Comment (fabianvf) [inline test/ansible-memcached/asserts.yml]: This needs to be `resource_name=...` instead of name, for dumb reasons.

**Comment by dymurray** [comment]
Cherry-picked this to v0.4.0 release as well.

https://github.com/operator-framework/operator-sdk/pull/972

**Comment by dymurray** [comment]
We are addressing all CI improvements here: https://github.com/operator-framework/operator-sdk/pull/985

File: test/ansible-memcached/asserts.yml [hunk 3]
```diff
@@ -45,6 +45,22 @@
...
+          kind=custom_resource.kind,
+          api_version=custom_resource.apiVersion,
+          namespace=namespace,
+          resource_name=custom_resource.metadata.name
+        )}}'
+
+    - name: Verify custom status exists
+      assert:
+        that: cr.resources[0].status.get("test") == "hello world"
+      when: cr is defined
```
[the remark this answers is not in the document]
Comment (dymurray) [reply test/ansible-memcached/asserts.yml]: See #985

File: test/ansible-memcached/asserts.yml [hunk 4]
```diff
@@ -45,6 +45,22 @@
           data:
             delete: me
 
+    - name: debug CR lookup
+      debug:
+        var: cr
+      vars:
+        deploy: '{{ lookup("k8s",
```
[the remark this answers is not in the document]
Comment (dymurray) [reply test/ansible-memcached/asserts.yml]: See #985

**Pull request merged**